staging/src/k8s.io/kubectl/pkg/cmd/config/delete_user.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+// DeleteUserOptions holds the data needed to run the command
+type DeleteUserOptions struct {
+	user string
+
+	configAccess clientcmd.ConfigAccess
+	config       *clientcmdapi.Config
+	configFile   string
+
+	genericclioptions.IOStreams
+}
+
+var (
+	deleteUserExample = templates.Examples(`
+		# Delete the minikube user
+		kubectl config delete-user minikube`)
+)
+
+// NewDeleteUserOptions creates the options for the command
+func NewDeleteUserOptions(ioStreams genericclioptions.IOStreams, configAccess clientcmd.ConfigAccess) *DeleteUserOptions {
+	return &DeleteUserOptions{
+		configAccess: configAccess,
```

Comment:
I'd move `configAccess` as a parameter to `Complete`, similarly to how we pass `Factory` in other commands. The reason for that is this will simplify testing, `IOStreams` are pretty easy to inject, it's a bit harder with `ConfigAccess`. `New*Options` should be more of a defaulting method.